active-record/src/main/java/com/iluwatar/active/record/MagicWand.java

```diff
@@ -0,0 +1,264 @@
+package com.iluwatar.active.record;
+
+import org.h2.tools.DeleteDbFiles;
+
+import java.sql.*;
+
+/**
+ * "Every single wand is unique and will depend for its character on the particular tree and magical creature
+ * from which it derives its materials. Moreover, each wand, from the moment it finds its ideal owner, will begin
+ * to learn from and teach its human partner." @ Garrick Ollivander
+ * <p/>
+ * Created by Stephen Lazarionok.
+ */
+public class MagicWand {
+
+    private Long id;
+
+    private Double lengthInches;
+
+    private WandWoodType wood;
+
+    private WandCoreType core;
+
+    public Long getId() {
+        return id;
+    }
+
+    public Double getLengthInches() {
+        return lengthInches;
+    }
+
+    public WandWoodType getWood() {
+        return wood;
+    }
+
+    public WandCoreType getCore() {
+        return core;
+    }
+
+    public void setLengthInches(Double lengthInches) {
+        this.lengthInches = lengthInches;
+    }
+
+    public void setWood(WandWoodType wood) {
+        this.wood = wood;
+    }
+
+    public void setCore(WandCoreType core) {
+        this.core = core;
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("MagicWand(");
+        sb.append(this.getWood());
+        sb.append(", ");
+        sb.append(this.getCore());
+        sb.append(", ");
+        sb.append(this.getLengthInches());
+        sb.append(" inch(es))");
+        return sb.toString();
+    }
+
+    /**
+     * ***********************************
+     * ****** Data access methods
+     * ***********************************
+     */
+
+    private static final String SELECT_SQL = "select * from wand where id = ?";
+    private static final String DELETE_SQL = "delete from wand where id = ?";
+    private static final String UPDATE_SQL = "update wand set length_inches = ?, wood = ?, core = ? where id = ?";
+    private static final String CREATE_SQL = "insert into wand values(?, ?, ?, ?)";
+
+    static {
```

Comment:
The alternative here is to use something like Spring IoC, make DB as a single bean and wrap it with the interface. However it will make an example more complicated. I want to keep it is very simple. The same approach was used by Martin Fowler to show Active Record pattern.